src/web/quick-start.md

```diff
@@ -0,0 +1,250 @@
+JavaScript mappings and potentially other handy files:
+[wasm-pack](https://github.com/rustwasm/wasm-pack) and
+[cargo-make](https://github.com/sagiegurari/cargo-make).
+
+`wasm-pack` lets you not only target the web natively but allows
+interoperability with [Node.js](https://nodejs.org/en/) or otherwise use along
+any JavaScript packages in a workflow such as
+[webpack](https://webpack.js.org/). The downside is that you must adhere to a
+certain project structure, and there is no built-in way to quickly serve a
+webpage for development and testing.
+
+`cargo-make` is a task runner that lets you define and configure a set of tasks
+in a script to execute upon building. It can not only be used to build for the
+web and generate the necessary JavaScript files, but even to quickly
+serve a webpage for development and testing. The downside here is that we must
+define all this behavior ourselves by creating a build script. There are
+templates you can use, but these are opinionated and you will have to follow
+their project structure then, as well.
+
+### Using `wasm-pack`
+
+First, you will need to install `wasm-pack`. You can do this with the help of
+cargo:
+
+```shell
+cargo install wasm-pack
+```
+
+Next, you will have to restructure your project as a library. This is more
+straightforward than it sounds. Simply rename your `main.rs` to `lib.rs` and
+rename your main function as follows:
+
+```rust
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn run() {
+    // Your code.
+}
+```
+
+You still require a `main.rs` if you desire to also run your code natively
+(i.e. on Windows, MacOS and/or Linux), but all that needs to be in it is a call
+to your main library function:
+
+```rust
+fn main() {
+    mylib::run();
+}
+
+```
+
+Finally, you will need the following additions in `Cargo.toml`:
+
+```toml
+[lib]
+crate-type = ["cdylib", "rlib"]
+
+[dependencies]
+wasm-bindgen = "0.2"
+```
+
+You can then simply build your project for the web by issuing
+`wasm-pack build --target web --release`. The resulting files will be in the
+`pkg/` directory.
+
+### Using `cargo-make`
+
+To actually get `cargo-make` building for the web, you need a build file.
+Writing this file is out of scope for this guide, but luckily the author of
+`bevy_webgl2` has provided [bevy_webgl2_app_template](https://github.com/mrk-its/bevy_webgl2_app_template) which you can use. If you are starting your
+project from scratch you can use the full template, otherwise you will need at
+least `Makefile.toml` and `index.html` from that repository.
+
+This template should allow for a project that runs both natively and on the web
+(but do read [Multi-Target](./web/multi-target.md), also), but you'll need
+to adhere to a specific project setup.
+
+First, use Cargo to install `cargo-make`:
+
+```shell
+cargo install cargo-make
+```
+
+Next, make sure your `Cargo.toml` looks like this:
+
+```toml
+[features]
+default = [
```

Comment:
Minor suggestion but you might want to mention adding `console_error_panic_hook` or else its going to be almost impossible for the user to debug what is going wrong in wasm if it *does* run int the browser but there is some non wasm compatible code.

Reply:
I'm trying to keep the code snippets as atomic as possible and relevant to their specific section. If I add it to the `cargo-make` instructions, I'll also have to add it to the `wasm-pack` instructions. I suspect the key problem here is discoverability, which may be better solved by simply referring to the problem in the beginning of the quickstart guide instead?